packages/SwingSet/src/vats/vat-admin/vat-vat-admin.js

```diff
@@ -41,88 +169,114 @@ export function buildRootObject(vatPowers) {
       }
       pendingBundles.get(bundleID).resolve(bundlecap);
       pendingBundles.delete(bundleID);
+      checkQuiescence();
     }
   }
 
-  function makeMeter(remaining, threshold) {
-    Nat(remaining);
-    Nat(threshold);
-    const meterID = D(vatAdminNode).createMeter(remaining, threshold);
-    const { updater, notifier } = makeNotifierKit();
-    const meter = Far('meter', {
-      addRemaining(delta) {
-        D(vatAdminNode).addMeterRemaining(meterID, Nat(delta));
-      },
-      setThreshold(newThreshold) {
-        D(vatAdminNode).setMeterThreshold(meterID, Nat(newThreshold));
-      },
-      get: () => D(vatAdminNode).getMeter(meterID), // returns BigInts
-      getNotifier: () => notifier,
-    });
-    meterByID.set(meterID, harden({ meter, updater }));
-    meterIDByMeter.set(meter, meterID);
-    return meter;
+  function pauseService() {
+    if (!quiescentP) {
+      [quiescentP, quiescentRR] = producePRR();
+    }
+    checkQuiescence();
+    return quiescentP;
   }
 
-  function makeUnlimitedMeter() {
-    const meterID = D(vatAdminNode).createUnlimitedMeter();
-    const { updater, notifier } = makeNotifierKit();
-    const meter = Far('meter', {
-      addRemaining(_delta) {},
-      setThreshold(_newThreshold) {},
-      get: () => harden({ remaining: 'unlimited', threshold: 0 }),
-      getNotifier: () => notifier, // will never fire
-    });
-    meterByID.set(meterID, harden({ meter, updater }));
-    meterIDByMeter.set(meter, meterID);
-    return meter;
+  function resumeService() {
+    if (quiescentP) {
+      quiescentRR.reject(Error('shutdown cancelled'));
+      quiescentP = null;
+      quiescentRR = null;
+    }
   }
 
-  function finishVatCreation(vatID) {
-    const [promise, pendingRR] = producePRR();
-    pending.set(vatID, pendingRR);
+  function finishMeter({ state, self }) {
+    meterByID.init(
+      state.meterID,
+      // harden({ meter: self, updater: state.updater }), // XXX RESTORE
+      harden({ meter: self }), // XXX TEMP
+    );
+    meterIDByMeter.set(self, state.meterID);
+    meterNotifiersTemp.set(self, makeNotifierKit()); // XXX TEMP
+  }
 
-    const [doneP, doneRR] = producePRR();
-    running.set(vatID, doneRR);
-    doneP.catch(() => {}); // shut up false whine about unhandled rejection
+  function insistServiceAvailable() {
+    assert(!quiescentP, 'service not available, try again later');
```

Comment:
I changed the name to `insistVatAdminServiceNotPaused`.  Suspenders _and_ belt.

I'm not sure that `throw Error(...)` goes very far in making the case that it's not an error.
I did angst a bit about using `assert` here when I wrote it, but our coding convention lately has been to eschew `throw` in favor of `assert`, so I bit the bullet and did that.  But it does seems like we should have a better convention for "abort processing due to exceptional circumstances that are not really erroneous", since such exceptional cases do seem to come up from time to time.